samples/batch/transcription-enabled-storage/CreateTranscription/CreateTranscription.cs

```diff
@@ -0,0 +1,295 @@
+﻿// <copyright file="CreateTranscription.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace CreateTranscriptionFunction
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Connector;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class CreateTranscription
+    {
+        [FunctionName("CreateTranscription")]
+        public static void Run([ServiceBusTrigger("create_transcription_queue", Connection = "AzureServiceBus")]Message message, ILogger log)
+        {
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            log.LogInformation($"C# ServiceBus queue trigger function processed message: {message.Label}");
+            log.LogInformation($"Received message: SequenceNumber:{message.SystemProperties.SequenceNumber} Body:{Encoding.UTF8.GetString(message.Body)}");
+
+            var location = TranscribeMessage(message, log).Result;
+            log.LogInformation("Request Created, Guid: " + location);
+        }
+
+        internal static async Task<string> TranscribeMessage(Message message, ILogger log)
+        {
+            var transcriptionCreationTime = DateTime.UtcNow;
+            var subscriptionKey = CreateTranscriptionEnvironmentVariables.AzureSpeechServicesKey;
+            var subscriptionRegion = CreateTranscriptionEnvironmentVariables.AzureSpeechServicesRegion;
+            var locationUri = new Uri($"https://{subscriptionRegion}.api.cognitive.microsoft.com");
+            var subscription = new Subscription(subscriptionKey, locationUri);
+
+            var messageBody = Encoding.UTF8.GetString(message.Body);
+            var busMessage = JsonConvert.DeserializeObject<ServiceBusMessage>(messageBody);
```

Comment:
It seems you only use busMessage, why not just pass busMessage?

Reply:
Because I need the Message in the exception handling if we need to reenqueue the initial service bus message. Because otherwise I'd need to create a new Message out of the ServiceBusMessage - which will just be unnecessarily complicated :D